File: 2nd_quarter/week02/Royce/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    
    # alp = []
    # 키를 최소 몇 번 눌러야 그 문자열을 작성할 수 있는지
    
#     // A,B,C,D에 대해 alp에 인덱스를 다 넣는다..?
#     for i in len(keymap):
#         for j in len(keymap[i]):
            
    for word in targets:
        num = 0
        
        for k in word:
            check = False
            idx = 100
            
            for h in keymap:
                if k in h:
                    idx = min(h.index(k)+1, idx)
                    check = True
            
            if not check:
                num = -1
                break
                # 불가능 하다는 뜻
    
            num += idx
        
        answer.append(num)
    
        
    return answer

File: 2nd_quarter/week02/Whaeun/대충만든 자판.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <map>

using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer;
    map<char, int> dic;
    
    for(int i = 0; i < keymap.size(); i++){
        for(int j = 0; j < keymap[i].size(); j++){
            if(dic[keymap[i][j]] != 0 && dic[keymap[i][j]] > j+1){
                dic[keymap[i][j]] = j + 1;
            }
            else if(dic[keymap[i][j]] == 0 ){
                dic[keymap[i][j]] = j + 1;
            }
        }
    }
    
    for(int i = 0; i < targets.size(); i++){
        int cnt = 0;
        bool check = true;
        
        for(int j = 0; j < targets[i].size(); j++){
            if(dic[targets[i][j]] == 0){
                check = false;
            }
            else{
                cnt += dic[targets[i][j]];
            }
        }
        
        if(!check){
            answer.push_back(-1);
        }
        else{
            answer.push_back(cnt);
        }
    }
    
    
    return answer;
}
